customComp/generatedForm.js

```diff
@@ -0,0 +1,47 @@
+webix.protoUI({
+    name: "generatedForm",
+    defaults: {
+        autoheight: true
+    },
+    $init: function (config) {
+        const cancelAction = config.cancelAction || function () {
```

Comment:
on config and events I'd do this something like that:
$init: function (config, events) {
        const baseEvents ={
cancelAction:function () { webix.message("Cancel");},
saveAction: function() {webix.message("Save");}
}
//merge to objects, e.g.:
Object.assign(baseEvents, events); 

//later
click: baseEvents.cancelAction 
either 
click: baseEvents.saveAction